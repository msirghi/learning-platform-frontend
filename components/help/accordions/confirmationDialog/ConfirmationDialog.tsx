import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { useTranslation } from '../../../../i18n';

type Props = {
  open: boolean;
  handleClose: () => void;
  searchInputValue: string;
  onDialogSubmit: () => void;
};

export const QuestionConfirmationDialog: React.FC<Props> = ({
  open,
  handleClose,
  searchInputValue,
  onDialogSubmit
}) => {
  const { t } = useTranslation();

  return (
    <div>
      <Dialog
        maxWidth='lg'
        open={open}
        onClose={handleClose}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'
      >
        <DialogTitle id='alert-dialog-title'>{t('help:confirmationDialog.title')}</DialogTitle>
        <DialogContent>
          <DialogContentText id='alert-dialog-description'>
            <div>
              <div>{t('help:confirmationDialog.messageWillBeSentTitle')}:</div>
              <div>
                <b>{searchInputValue}</b>
              </div>
              <div>{t('help:confirmationDialog.confirmYourAction')}</div>
            </div>
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button variant='outlined' onClick={handleClose} color='secondary'>
            {t('help:confirmationDialog.cancelButtonLabel')}
          </Button>
          <Button variant='outlined' onClick={onDialogSubmit} color='primary'>
            {t('help:confirmationDialog.confirmButtonLabel')}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};
