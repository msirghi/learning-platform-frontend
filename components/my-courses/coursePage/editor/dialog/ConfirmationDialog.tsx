import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { useTranslation } from '../../../../../i18n';

type Props = {
  message: string;
  onSubmit: () => void;
  onCancel: () => void;
  open: boolean;
};

export const ConfirmationDialog: React.FC<Props> = ({ open, message, onSubmit, onCancel }) => {
  const { t } = useTranslation();

  return (
    <div>
      <Dialog
        open={open}
        onClose={onCancel}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'
      >
        <DialogTitle id='alert-dialog-title'>{t('courses:confirmationDialogTitle')}</DialogTitle>
        <DialogContent>
          <DialogContentText id='alert-dialog-description'>{message}</DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={onCancel} color='secondary' variant='outlined'>
            {t('courses:cancel')}
          </Button>
          <Button onClick={onSubmit} color='primary' variant='outlined'>
            {t('courses:continue')}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};
