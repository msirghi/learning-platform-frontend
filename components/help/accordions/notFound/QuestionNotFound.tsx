import { Button } from '@material-ui/core';
import React, { useState } from 'react';
import styles from './QuestionNotFound.module.scss';
import { QuestionConfirmationDialog } from '../confirmationDialog/ConfirmationDialog';
import { useSnackbar } from 'notistack';
import { useTranslation } from '../../../../i18n';

type Props = {
  searchInputValue: string;
};

export const QuestionNotFound: React.FC<Props> = ({ searchInputValue }) => {
  const [confirmationDialogOpen, setConfirmationDialogOpen] = useState<boolean>(false);
  const { enqueueSnackbar } = useSnackbar();
  const { t } = useTranslation();

  const toggleConfirmationDialog = (val: boolean) => setConfirmationDialogOpen(val);

  const onDialogSubmit = () => {
    toggleConfirmationDialog(false);
    enqueueSnackbar(t('help:messageSentSuccessMessage'), { variant: 'success' });
  };

  return (
    <>
      <QuestionConfirmationDialog
        open={confirmationDialogOpen}
        onDialogSubmit={onDialogSubmit}
        handleClose={() => toggleConfirmationDialog(false)}
        searchInputValue={searchInputValue}
      />
      <div className={styles.notFoundContainer}>
        <div>{t('help:nothingWasFoundLabel')}</div>

        <div className={styles.buttonContainer}>
          <Button variant='outlined' color='primary' onClick={() => toggleConfirmationDialog(true)}>
            {t('help:sendMessageButtonLabel')}
          </Button>
        </div>
      </div>
    </>
  );
};
