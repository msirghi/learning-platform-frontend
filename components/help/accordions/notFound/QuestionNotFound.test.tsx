import { Button } from '@material-ui/core';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SnackbarProvider } from 'notistack';
import React from 'react';
import { QuestionConfirmationDialog } from '../confirmationDialog/ConfirmationDialog';
import { QuestionNotFound } from './QuestionNotFound';

describe('QuestionNotFound component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <SnackbarProvider>
        <QuestionNotFound searchInputValue='val' />
      </SnackbarProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should toggle confirmation dialog on button click', () => {
    const wrapper = mount(
      <SnackbarProvider>
        <QuestionNotFound searchInputValue='val' />
      </SnackbarProvider>
    );
    const button = wrapper.find(Button);
    expect(button).toHaveLength(1);

    button.simulate('click');
    const dialog = wrapper.find(QuestionConfirmationDialog);
    expect(dialog).toHaveLength(1);
    expect(dialog.props().open).toBeTruthy();
  });

  it('should close confirmation dialog on submitting', () => {
    const wrapper = mount(
      <SnackbarProvider>
        <QuestionNotFound searchInputValue='val' />
      </SnackbarProvider>
    );
    const dialog = wrapper.find(QuestionConfirmationDialog);
    expect(dialog).toHaveLength(1);

    dialog.props().onDialogSubmit();
    expect(dialog.props().open).toBeFalsy();
  });
});
