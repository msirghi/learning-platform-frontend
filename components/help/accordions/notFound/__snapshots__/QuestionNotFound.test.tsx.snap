// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionNotFound component should match the snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "closeSnackbar": [Function],
      "enqueueSnackbar": [Function],
    }
  }
>
  <QuestionNotFound
    searchInputValue="val"
  />
</ContextProvider>
`;
